MatrixSDK/Crypto/MXCryptoV2.swift

```diff
@@ -101,33 +158,119 @@ public class MXCryptoV2: MXCrypto {
         warnNotImplemented()
     }
     
-    public override func encryptEventContent(_ eventContent: [AnyHashable : Any]!, withType eventType: String!, in room: MXRoom!, success: (([AnyHashable : Any]?, String?) -> Void)!, failure: ((Error?) -> Void)!) -> MXHTTPOperation! {
-        warnNotImplemented()
-        return nil
+    // MARK: - Encrypt / Decrypt
+    
+    public override func encryptEventContent(
+        _ eventContent: [AnyHashable : Any]!,
+        withType eventType: String!,
+        in room: MXRoom!,
+        success: (([AnyHashable : Any]?, String?) -> Void)!,
+        failure: ((Error?) -> Void)!
+    ) -> MXHTTPOperation! {
+        guard let content = eventContent, let eventType = eventType, let roomId = room.roomId else {
+            MXLog.debug("[MXCryptoV2] encryptEventContent: Missing data to encrypt")
+            return nil
+        }
+        
+        guard isRoomEncrypted(roomId) else {
+            MXLog.error("[MXCryptoV2] encryptEventContent: attempting to encrypt event in room without encryption")
+            return nil
+        }
+        
+        MXLog.debug("[MXCryptoV2] encryptEventContent: Encrypting content")
+        
+        Task {
+            do {
+                let users = try await getRoomUserIds(for: room)
+                let result = try await machine.encrypt(
+                    content,
+                    roomId: roomId,
+                    eventType: eventType,
+                    users: users
+                )
+                
+                await MainActor.run {
+                    success?(result, kMXEventTypeStringRoomEncrypted)
+                }
+            } catch {
+                MXLog.error("[MXCryptoV2] encryptEventContent: Error encrypting content - \(error)")
+                await MainActor.run {
+                    failure?(error)
+                }
+            }
+        }
+        return MXHTTPOperation()
     }
     
     public override func hasKeys(toDecryptEvent event: MXEvent!, onComplete: ((Bool) -> Void)!) {
         warnNotImplemented()
     }
     
     public override func decryptEvent(_ event: MXEvent!, inTimeline timeline: String!) -> MXEventDecryptionResult! {
-        warnNotImplemented()
-        return nil
+        guard let event = event else {
+            MXLog.debug("[MXCryptoV2] Missing event")
+            return nil
+        }
+        do {
+            return try machine.decryptEvent(event)
+        } catch {
+            MXLog.error("[MXCryptoV2] decryptEvent: \(error)")
+            let result = MXEventDecryptionResult()
+            result.error = error
```

Comment:
we will need a way to map this to existing errors (will help to be consistent with existing posthog events)

Reply:
Adding this to my TODOs. Given that I've merged Sentry now, these will be primarily tracked there, and after implementing further [error details](https://github.com/matrix-org/matrix-rust-sdk/issues/734), we can observe the tracked data and decide how to map to posthog